CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntax/SyntaxNodeFile.swift

```diff
@@ -0,0 +1,359 @@
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+import SwiftSyntaxBuilder
+import SyntaxSupport
+import Utils
+
+/// This gyb-file generates the syntax nodes for SwiftSyntax. To keep the generated
+/// files at a managable file size, it is to be invoked multiple times with the
+/// global variable EMIT_KIND set to a base kind listed in
+/// gyb_syntax_support/kinds.py:SYNTAX_BASE_KINDES. It then only emits those syntax
+/// nodes whose base kind are that specified kind.
+func syntaxNode(emitKind: String) -> SourceFileSyntax {
+  SourceFileSyntax(leadingTrivia: .docLineComment(generateCopyrightHeader(for: "generate-swiftsyntax"))) {
+    for node in SYNTAX_NODES {
+      if node.isBase {
+        // Handled in SyntaxNodesBase.swift.gyb
+      } else if !node.collectionElement.isEmpty {
+        // Handled in SyntaxCollections.swift.gyb
+      } else if node.baseKind != emitKind {
+        // Emit node
+      } else {
+        // We are actually handling this node now
+        let nodeDoc = node.description.map { "/// \($0)" }
+        try! StructDeclSyntax("""
+        // MARK: - \(raw: node.name)
+
+        \(raw: nodeDoc ?? "")
+        public struct \(raw: node.name): \(raw: node.baseType.syntaxBaseName)Protocol, SyntaxHashable
+        """) {
+          for child in node.children {
+            if !child.nodeChoices.isEmpty {
+              try! EnumDeclSyntax("public enum \(raw: child.name): SyntaxChildChoices") {
+                for choice in child.nodeChoices {
+                  DeclSyntax("case `\(raw: choice.swiftName)`(\(raw: choice.typeName))")
+                }
+
+                try! VariableDeclSyntax("public var _syntaxNode: Syntax") {
+                  try! SwitchStmtSyntax("switch self") {
+                    for choice in child.nodeChoices {
+                      SwitchCaseSyntax("case .\(raw: choice.swiftName)(let node):") {
+                        StmtSyntax("return node._syntaxNode")
+                      }
+                    }
+                  }
+                }
+
+                DeclSyntax("init(_ data: SyntaxData) { self.init(Syntax(data))! }")
+
+                for choice in child.nodeChoices {
+                  if let choiceNode = SYNTAX_NODE_MAP[choice.syntaxKind], choiceNode.isBase {
+                    DeclSyntax("""
+                        public init<Node: \(raw: choiceNode.name)Protocol>(_ node: Node) {
+                          self = .\(raw: choice.swiftName)(\(raw: choiceNode.name)(node))
+                        }
+                        """)
```

Comment:
Investing that swift-format doesn’t complain about this but the closing quote of this literal is indented by 4 spaces (same below) and the one after is not indented at all. I think I would expect 2 space of indentation.

Reply:
Fixed in #1301